src/tests/ftest/slurm_setup.py

```diff
@@ -206,26 +236,46 @@ def start_slurm(args):
     cmd_list = [
         "mkdir -p /var/log/slurm",
         "chown {}. {}".format(args.user, "/var/log/slurm"),
+        "mkdir -p /var/spool/slurmd",
+        "mkdir -p /var/spool/slurmctld",
         "mkdir -p /var/spool/slurm/d",
         "mkdir -p /var/spool/slurm/ctld",
-        "chown {}. {}/ctld".format(args.user, "/var/spool/slurm")
+        "chown {}. {}/ctld".format(args.user, "/var/spool/slurm"),
+        "chown {}. {}".format(args.user, "/var/spool/slurmctld"),
+        "chmod 775 {}".format("/var/spool/slurmctld"),
+        "rm -f /var/spool/slurmctld/clustername"
         ]
 
     if execute_cluster_cmds(all_nodes, cmd_list, args.sudo) > 0:
         return 1
 
-    # Startup the slurm control service
-    if execute_cluster_cmds(args.control, SLURMCTLD_STARTUP, args.sudo) > 0:
+    # Startup the slurm service
+    status = execute_cluster_cmds(all_nodes, SLURMD_STARTUP, args.sudo)
+    if status > 0 or args.debug:
+        execute_cluster_cmds(all_nodes, SLURMD_STARTUP_DEBUG, args.sudo)
+    if status > 0:
         return 1
 
-    # Startup the slurm service
-    if execute_cluster_cmds(all_nodes, SLURMD_STARTUP, args.sudo) > 0:
+    # Startup the slurm control service
+    status = execute_cluster_cmds(args.control, SLURMCTLD_STARTUP, args.sudo)
+    if status > 0 or args.debug:
+        execute_cluster_cmds(args.control, SLURMCTLD_STARTUP_DEBUG, args.sudo)
+    if status > 0:
         return 1
+    # wait until the ctl node updates the slurm state from unknown to idle
+    sleep(20)
```

Comment:
We should be using a more deterministic approach to determine if the code should progress.  Either query some state that slurm or systemctl reports that would indicate slurmctld is ready or use a loop to retry a command until it is successful or times out - here a `sleep(1)` would be acceptable between retries.